newIDE/app/src/SceneEditor/index.js

```diff
@@ -486,15 +482,15 @@ export default class SceneEditor extends React.Component<Props, State> {
   };
 
   _onAddInstanceUnderCursor = () => {
-    if (!this.state.selectedObjectNames.length || !this.editor) {
+    if (!this.state.selectedObjects.length || !this.editor) {
       return;
     }
 
-    const objectSelected = this.state.selectedObjectNames[0];
+    const objectSelected = this.state.selectedObjects[0];
     const cursorPosition = this.editor.getLastCursorSceneCoordinates();
-    this._addInstance(cursorPosition, objectSelected);
+    this._addInstance(cursorPosition, objectSelected.object.getName());
```

Comment:
It feels a bit weird to read this: `objectSelected.object`. I think you should go with the slightly heavier but clearer naming `selectedObjectsWithContext`